xpartition.py

```diff
@@ -637,3 +652,56 @@ def write(self, rank):
     def __iter__(self):
         self._initialize_store()
         return iter(range(len(self.plan.input_partition)))
+
+
+def get_chunks_encoding(da: xr.DataArray) -> Tuple[int, ...]:
+    return tuple(chunk_size for chunk_size, *_ in da.chunks)
```

Comment:
this is a bit cryptic to me, I thought `xr.DataArray.chunks` already returned a tuple of chunk sizes.

Reply:
Oh, I guess dask can have multiple chunk sizes for each dim? But we can just use the first since an error will be raised elsewhere if the dask chunks don't conform to zarr's expectations (i.e. all constant except for last)?

Reply:
That's correct, `da.chunks` returns a tuple of tuples and we only want the first element from each, assuming the chunks are valid for zarr.  Indeed though we should validate that here.  I added code / error messages based on [what xarray uses for this](https://github.com/pydata/xarray/blob/282235f4f3e3432c9defaee45777ecef256d8684/xarray/backends/zarr.py#L248-L269).  

I also took the opportunity to more formally split out code that was adapted from xarray into its own module (`xarray_utils.py`) with a copy of the xarray license.